pymatgen/io/vasp/sets.py

```diff
@@ -1044,6 +1044,185 @@ def __init__(self, structure, potim=0.015, **kwargs):
         self.config_dict["INCAR"].pop("NPAR", None)
 
 
+class MVLGWscDFTSet(DictSet):
+    """
+    Standard DFT Static run for a nonscf exact diagonalization calc in GW calc.
+    """
+    CONFIG = loadfn(os.path.join(MODULE_DIR, "MVLGWSet.yaml"))
+
+    def __init__(self, structure, potcar_functional="PBE_54",
+                 reciprocal_density=100, **kwargs):
+        super(MVLGWscDFTSet, self).__init__(
+            structure, MVLGWscDFTSet.CONFIG, **kwargs)
+        self.potcar_functional = potcar_functional
+        self.reciprocal_density = reciprocal_density
+        self.kwargs = kwargs
+
+    @property
+    def kpoints(self):
+        """
+        Generate gamma center mesh k-points grid for GW calc.
+        :return: gamma centered k-points
+        """
+        return Kpoints.automatic_density_by_vol(self.structure,
+                                                self.reciprocal_density,
+                                                force_gamma=True)
+
+
+class MVLGWDiagDFTSet(MVLGWscDFTSet):
+    """
+    Non self-consistent exact diagonalization step for a GW run,
+    which is the 2nd step-2 for a GW/BSE run.
+    Usage: use from_prev_calc method for creating inputs.
+
+    Tips: In this step you need to do convergence testing for NBANDS.
+    """
+    def __init__(self, structure, prev_incar=None, **kwargs):
+        super(MVLGWDiagDFTSet, self).__init__(structure, **kwargs)
+        self.kwargs = kwargs
+        self.prev_incar = prev_incar
+
+    @property
+    def incar(self):
+        parent_incar = super(MVLGWDiagDFTSet, self).incar
+        incar = Incar(self.prev_incar) if self.prev_incar is not None else \
+            Incar(parent_incar)
+
+        incar.update({"ALGO": "Exact", "NELM":1, "LOPTICS": True,
+                      "LPEAD": True, "OMEGAMAX": 40})
+
+        return incar
+
+    @classmethod
+    def from_prev_calc(cls, prev_calc_dir, copy_wavecar=True,
+                       nbands_factor=5, **kwargs):
+        """
+
+        :param prev_calc_dir:
```

Comment:
These should be Google style docs. You can copy and paste from MPStaticSet to save time.